lib/svg_sprite/svg_sprite.rb

```diff
@@ -351,10 +351,16 @@ def self.bundle(theme_id = nil)
     end
 
     custom_svg_sprites(theme_id).each do |item|
-      svg_file = Nokogiri::XML(item[:sprite]) do |config|
-        config.options = Nokogiri::XML::ParseOptions::NOBLANKS
+      begin
+        svg_file = Nokogiri::XML(item[:sprite]) do |config|
+          config.options = Nokogiri::XML::ParseOptions::NOBLANKS
+        end
+      rescue
+        Rails.logger.warn("Bad XML in custom sprite in theme with ID=#{theme_id}.")
```

Comment:
A couple of points here:

1. I think the log message here is a little too generic and doesn't tell what the problem is exactly. Can we include the type/class and message (if any) of the rescued error in the log message here?

2. I wonder if we should also validate the XML/sprite file and surface errors earlier when it's uploaded? I think right now (correct me if I'm wrong) the app doesn't perform any checks on sprite files bundled with themes, and so when someone uploads a broken sprite file, Discourse accepts it but it won't have any effects because it's broken and they'd have to remember to check `/logs` to figure out what's wrong. Obviously this needs a fair amount of work, so feel free to merge this PR now and we can improve this later.

Reply:
Thanks @OsamaSayegh, I have made both adjustments. In the UI, admins that upload a broken sprite will see this: 

<img width="796" alt="image" src="https://user-images.githubusercontent.com/368961/125673405-5e5f511d-ac83-4cf3-a85c-7f18d4cfe525.png">

Reply:
Perfect, thanks you so much!